### DISCUSSION TITLE
Is there a way to set a lifetime/duration for an entry. So it expires after a while?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#19381] [Q&A and General discussion]
Is there a way to set a lifetime/duration for an entry. So it expires after a while?

Or maybe something like search by date?

My goal is to temporality store recently added vectors,  and if I have a similar enough vector that not older say 5 minutes ignore the query, so I won't use recent events.

Or is it better use something like reddis and check similarity manually?

**Comment by yhmo** [comment]
The feature "expire legacy data" is in developing, might be available on v2.2.
The feature "search with time travel" may be helpful to you: https://milvus.io/docs/v2.1.x/timetravel.md
